pkg/controllers/dlb/controller.go

```diff
@@ -92,8 +97,13 @@ func (c *controller) UpdateDaemonSet(rawObj client.Object, ds *apps.DaemonSet) (
 		updated = true
 	}
 
-	if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, dp.Spec.NodeSelector) {
-		ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector
+	if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, dp.Spec.NodeSelector) &&
+		!reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, defaultNodeSelector) {
```

Comment:
This would prevent updating ds node selector when plugin nodeselector is not empty and is not equal to defaultNodeSelector

Reply:
btw, as this piece of code is repeated for all controllers we may want to consider an api code instead of copying&pasting it everywhere.

Reply:
@bart0sh About the matter of making api call, what do you think about "defaultNodeSelector" variable? Though the value is same as map[string]string{"kubernetes.io/arch": "amd64"}, actually it has the value from each device plugin's yaml file, as you know.!

Do you think it's better to get the value as a parameter of the api function
or just go back to put just map[string]string{"kubernetes.io/arch": "amd64"}
or another way?

Reply:
I like it as it is now. It's taken from the correspondent daemonset yaml. If we hardcode it as map[string]string{"kubernetes.io/arch": "amd64"} and then for some reason decide to change it we'd need to update it in two places: in the yaml and in the code, which is not what want, I believe.

Reply:
@bart0sh I see.! It was what I was also thinking.!